src/core/datatypes/trees/TopologyNode.cpp

```diff
@@ -216,6 +219,77 @@ void TopologyNode::addBranchParameters(std::string const &n, const std::vector<s
 }
 
 
+bool TopologyNode::hasNodeComment(const std::string& comment) const
+{
+    for(auto& node_comment: node_comments)
+        if (node_comment == comment)
+            return true;
+    return false;
+}
+
+std::optional<std::string> TopologyNode::getNodeParameter(const std::string& name) const
+{
+    for(auto& node_comment: node_comments)
+    {
+        if (node_comment.substr(0,name.size()) == name and node_comment.size() > name.size() and node_comment[name.size()] == '=')
+        {
+            return node_comment.substr(name.size()+1);
+        }
+    }
+
+    // Not found
+    return {};
+}
+
+bool TopologyNode::setNodeParameter(const std::string& name, const std::string& value)
```

Comment:
It always sets the value, but in the case where there is already a `name=value` comment, it replaces the existing comment instead of appending a second value for the name.

I had a comment in the header file, I'll add one here as well.